packages/ws/src/throttling/SimpleIdentifyThrottler.ts

```diff
@@ -0,0 +1,50 @@
+import { setTimeout as sleep } from 'node:timers/promises';
+import { Collection } from '@discordjs/collection';
+import { AsyncQueue } from '@sapphire/async-queue';
+import type { IIdentifyThrottler } from './IIdentifyThrottler';
+
+/**
+ * The state of a rate limit key's identify queue.
+ */
+export interface IdentifyState {
+	queue: AsyncQueue;
+	resetsAt: number;
+}
+
+/**
+ * Local, in-memory identify throttler.
+ */
+export class SimpleIdentifyThrottler implements IIdentifyThrottler {
+	private readonly states = new Collection<number, IdentifyState>();
+
+	public constructor(private readonly maxConcurrency: number) {}
+
+	/**
+	 * {@inheritDoc IIdentifyThrottler.waitForIdentify}
+	 */
+	public async waitForIdentify(shardId: number): Promise<void> {
+		const key = shardId % this.maxConcurrency;
+
+		const state = this.states.ensure(key, () => {
+			return {
+				queue: new AsyncQueue(),
+				resetsAt: Number.POSITIVE_INFINITY,
+			};
+		});
```

Comment:
```suggestion
		const state = this.states.ensure(key, () => ({
			queue: new AsyncQueue(),
			resetsAt: Number.POSITIVE_INFINITY,
		}));
```

Reply:
I prefer the current code ngl, but it's up to DD if they wanna move to your suggestion